### PR TITLE
Backports 0.18 pr13

### PR DESCRIPTION
**Pull request by Munkybooty**
This breaks p2p_connect_to_devnet and i'm not too sure as to why.

### CODE REVIEW

**Comment by Munkybooty** [comment]
i swear i pressed develop when i created the pr and yet somehow it switched back to master. weird...

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see https://github.com/UdjinM6/dash/commits/pr4333 + should drop 14670 (it was merged earlier)

**Review by UdjinM6 — APPROVED** [review]
utACK (verified that dash-qt compiles and runs fine with qt 5.9.7 on local macos, no visual issues)

File: test/functional/mining_basic.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
     P2PDataStore,
 )
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.script import CScriptNum
```
Comment (PastaPastaPasta) [inline test/functional/mining_basic.py:23]: 14816 this should be below 
```
    bytes_to_hex_str as b2x,
)
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: test/functional/mining_basic.py (hunk 1 above)
Comment (Munkybooty) [reply test/functional/mining_basic.py:23]: it should be on line 27

Comment (PastaPastaPasta) [reply test/functional/mining_basic.py:23]: I think. Just down a few lines. Below
```

from test_framework.util import (
    assert_equal,
    assert_raises_rpc_error,
    bytes_to_hex_str as b2x,
)
```

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Pull request merged**